### PR TITLE
 	Fix client side amount handling

### PR DESCRIPTION
**Pull request by gbirke**
Move amount update and event handling into a component to make it more
consistent with the rest of the form code and make store updates work
across pages.

Always display the amount in the text field.

### CODE REVIEW

File: app/js/tests/test_form_components.js [hunk 1]
```diff
@@ -65,3 +65,88 @@ test( 'Rendering the radio component sets the value as array', function ( t ) {
...
+	t.ok( selectElement.val.callCount === 0, 'select element value is not set' );
+	t.ok( selectElement.prop.calledOnce, 'property was set' );
+	t.ok( selectElement.prop.calledWith( 'checked', false ), 'check property was removed' );
+
+	component.render( { amount: '50,00', isCustomAmount: false } );
+	t.ok( inputElement.val.calledTwice, 'value is set again' );
+	t.ok( inputElement.val.calledWith( '50,00' ) );
+	t.ok( selectElement.val.calledOnce, 'select element value is set' );
+	t.ok( selectElement.val.calledWith( [ '50,00' ] ), 'select element value is set' ); // needs to be array for selects
+	t.end();
```
Comment (JeroenDeDauw) [inline app/js/tests/test_form_components.js]: Are those not two different tests?

**Comment by JeroenDeDauw** [comment]
+1 to 1618045fe67cde79e9060d7a5a9f0ca7bbee938a

File: app/js/tests/test_form_components.js (hunk 1 above)
Comment (gbirke) [inline app/js/tests/test_form_components.js]: Could be considered two different tests. Will refactor that.

**Comment by gbirke** [comment]
Splitted the test as requested by @JeroenDeDauw and rebased to master.

**Pull request merged**